apps/files/utils.py

```diff
@@ -91,49 +91,77 @@ class Extractor(object):
     def parse(cls, path):
         install_rdf = os.path.join(path, 'install.rdf')
         package_json = os.path.join(path, 'package.json')
+        manifest_json = os.path.join(path, 'manifest.json')
         if os.path.exists(install_rdf):
             return RDFExtractor(path).data
         elif os.path.exists(package_json):
             return PackageJSONExtractor(package_json).parse()
+        elif os.path.exists(manifest_json):
+            return ManifestJSONExtractor(manifest_json).parse()
         else:
-            raise forms.ValidationError("No install.rdf or package.json found")
+            raise forms.ValidationError(
+                "No install.rdf or package.json or manifest.json found")
 
 
-class PackageJSONExtractor(object):
-    def __init__(self, path):
+def get_appversions(app, min_version, max_version):
+    """Return the `AppVersion`s that correspond to the given versions."""
+    qs = AppVersion.objects.filter(application=app.id)
+    min_appver = qs.get(version=min_version)
+    max_appver = qs.get(version=max_version)
+    return min_appver, max_appver
+
+
+def get_simple_version(version_string):
+    """Extract the version number without the ><= requirements.
+
+    This simply extracts the version number without the ><= requirement so
+    it will not be accurate for version requirements that are not >=, <= or
+    = to a version.
+
+    >>> get_simple_version('>=33.0a1')
+    '33.0a1'
+    """
+    if not version_string:
+        return ''
+    return re.sub('[<=>]', '', version_string)
+
+
+class JSONExtractor(object):
+    def __init__(self, path, data=''):
         self.path = path
-        self.data = json.load(open(self.path))
+        self.data = json.loads(data) if data else json.load(open(self.path))
```

Comment:
Nope, we don't, it's already taken care of in https://github.com/mozilla/olympia/blob/master/apps/files/utils.py#L436 (it's in a `try/except` that will display a "could not parse install.rdf" if something went wrong).
But that's a good catch though, I need to change the error messages :)